### ISSUE TITLE
Unable to load flagged posts

### ISSUE THREAD

**Issue by lvwrence** [flarum/core#1916]
## Bug Report

**Current Behavior**
Clicking on the flagged posts dropdown loads forever and throws an error:

```
forum-444ec428.js:66 Uncaught TypeError: e.slug is not a function
    at Function.t.route.discussion (forum-444ec428.js:66)
    at Function.t.route.post (forum-444ec428.js:66)
    at forum-444ec428.js:565
    at Array.map (<anonymous>)
    at a.o.view (forum-444ec428.js:565)
    at a.e.render (forum-444ec428.js:1)
    at a.n.getMenu (forum-444ec428.js:66)
    at a.n.view (forum-444ec428.js:1)
    at a.e.render (forum-444ec428.js:1)
    at Object.o [as view] (forum-444ec428.js:1)
```

Related discussion: https://discuss.flarum.org/d/21521-flagged-posts-can-t-load

**Steps to Reproduce**
1. Flag a post as a user
2. As admin, click on flagged posts dropdown

**Expected Behavior**
The flagged posts dropdown should appear with all the flagged posts.

**Screenshots**
<img width="784" alt="Screen Shot 2019-10-24 at 2 09 50 PM" src="https://user-images.githubusercontent.com/963922/67525728-2340ed00-f668-11e9-9c51-9ab3c6e7d8c7.png">

**Environment**
- Flarum version: 0.1.0-beta.10
- Website URL: demo.flarumpro.com
- Webserver: Apache 2
- Hosting environment: Heroku
- PHP version: x.y.z
- Browser: Chrome 77

```
Output of "php flarum info", run this in terminal in your Flarum directory.
```

**Possible Solution**

It looks like app.route.discussion() is being called but the value of discussion is `false`. I am investigating further.

**Comment by lvwrence** [comment]
I believe the issue is essentially that the `post` argument in `app.route.post()` expects a `slug` attribute, which doesn't exist in the `flags` plugin API when fetching the flagged posts.

**Comment by datitisev** [comment]
Thanks for the report. This seems to occur when a discussion is deleted from the DB but its posts are not (caused by #1909).

#1910 will fix this, unless I'm completely wrong. 🙂.

Not sure if we want to keep this open or not, will let other core members decide.

**Comment by datitisev** [comment]
This *should've* been fixed by #1910 in beta 11.
